### ISSUE TITLE
Enable measurement of the same model used for serving.

### ISSUE THREAD

**Issue by BacharL** [vllm-project/llm-compressor#64]
Hi.
Currently the quantization process requires loading a model and checkpoint in llm-compressor, performing measurement, saving checkpoint and then loading this checkpoint in vLLM. In vLLM the model structure is different; linear layers are replaced with RowParallellLinear, attention implementation is different and kv cache loading and saving is different.

We would like to perform quantization of non-linear layers, for example activations of attention qk matmul and attention softmax. We implemented a custom attention in vllm which uses torch.nn.Matmul and torch.nn.Softmax instead of torch..nn.functional api and use Intel Neural Compressor to replace these modules with measurement modules.

Currently it is not possible to use llm-compressor for measurement of these layers since it loads huggingface model and not the model used in vllm.

For this reason our measurement process is currently performed in vLLM framework instead of llm-compressor

```
llm = LLM(model=...)
model = llm.llm_engine.model_executor.driver_worker.model_runner.model.model
inc.prepare(model)  # Converts the model loaded in the vLLM engine model_runner by replacing nn.Modules that support measurement/quantization
llm.generate(...)
inc.finalize_calibration(model)  # Save checkpoint and scales
```

We would like to know if you have plans to enable measurement of a model in it's vLLM structure?
We would like to avoid having to maintain 2 models, one for measurement and one for serving.

Also, vLLM currently loads quantized implementation only for linear layers, we seek the ability to replace non-linear layers after vllm during vllm model loading.

Thanks

**Comment by BacharL** [comment]
Thanks Robert.

Our goal was to measure activations of matmul and softmax inside the attention, for this purpose we created a custom attention implementation which replaced these operations with custom nn.Modules that perform the measurement. This change has to be done in the model code for measurement and also in attention implementation in vllm.
Remapping state during weight loading is possible just requires code specific to vllm. it would require saving these scales on the attention module instead of the matmul/softmax module, since these don't have state_dict and are not saved in the checkpoint.

I now understand that this change to the model is undesirable and we need to find a way to measure internal activations in the attention without it.
